src/main/java/org/terasology/ReplayTestingEnvironment.java

```diff
@@ -168,6 +204,13 @@ protected boolean isInitialised() {
         return this.isInitialised;
     }
 
-
-
+    protected void waitUntil(BooleanSupplier supplier) {
+        try {
+            while (!supplier.getAsBoolean()) {
+                Thread.sleep(1000);
+            }
+        } catch (Exception e) {
+            logger.error("Error has occurred in the waitUntil method: ", e);
```

Comment:
Should we really be catching and logging exceptions here? An uncaught exception will fail the test and signify that there may be an unwanted problem in the code executed in `supplier.getAsBoolean()`. If the exception is caught and logged here, it might allow problems to slip under the radar. As a rule, I never catch exceptions in tests unless I _want_ that exception to occur (or want to manually fail the test via `Assert.fail`).